src/transactions/test/RevokeSponsorshipTests.cpp

```diff
@@ -579,6 +580,90 @@ TEST_CASE("update sponsorship", "[tx][sponsorship]")
                     checkSponsorship(ltx, a2, 0, nullptr, 0, 2, 1, 0);
                     ltx.commit();
                 }
+
+                SECTION("data")
+                {
+                    DataValue value;
+                    std::string dataName = "test";
+                    auto a1 = root.create("a1", minBal(1));
+                    auto a2 = root.create("a2", minBal(2));
+
+                    auto tx1 = transactionFrameFromOps(
+                        app->getNetworkID(), root,
+                        {root.op(beginSponsoringFutureReserves(a1)),
+                         a1.op(manageData(dataName, &value)),
+                         a1.op(endSponsoringFutureReserves())},
+                        {a1});
+
+                    LedgerTxn ltx(app->getLedgerTxnRoot());
+                    TransactionMeta txm1(2);
+                    REQUIRE(tx1->checkValid(ltx, 0, 0, 0));
+                    REQUIRE(tx1->apply(*app, ltx, txm1));
+
+                    auto tx2 = transactionFrameFromOps(
+                        app->getNetworkID(), root,
+                        {a2.op(beginSponsoringFutureReserves(root)),
+                         root.op(revokeSponsorship(dataKey(a1, dataName))),
+                         root.op(endSponsoringFutureReserves())},
+                        {a2});
+
+                    TransactionMeta txm2(2);
+                    REQUIRE(tx2->checkValid(ltx, 0, 0, 0));
+                    REQUIRE(tx2->apply(*app, ltx, txm2));
+
+                    checkSponsorship(ltx, dataKey(a1, dataName), 1,
+                                     &a2.getPublicKey());
+                    checkSponsorship(ltx, a1, 0, nullptr, 1, 2, 0, 1);
+                    checkSponsorship(ltx, a2, 0, nullptr, 0, 2, 1, 0);
+                    checkSponsorship(ltx, root, 0, nullptr, 0, 2, 0, 0);
+                    ltx.commit();
+                }
+
+                SECTION("offer")
+                {
+                    auto cur1 = makeAsset(root, "CUR1");
+                    auto native = makeNativeAsset();
+                    auto a1 = root.create("a1", minBal(3));
+                    auto a2 = root.create("a2", minBal(2));
+
+                    auto toSet = static_cast<uint32_t>(AUTH_REQUIRED_FLAG) |
+                                 static_cast<uint32_t>(AUTH_REVOCABLE_FLAG);
+                    root.setOptions(txtest::setFlags(toSet));
```

Comment:
Any particular reason you made the asset authorization required?

Reply:
No good catch. Looks like copy paste sloppiness. Fixed.